internal/types/payment_gateway.go

```diff
@@ -0,0 +1,102 @@
+					PaymentGatewayTypeStripe,
+					PaymentGatewayTypeRazorpay,
+					PaymentGatewayTypeFinix,
+				},
+			}).
+			Mark(ierr.ErrValidation)
+	}
+}
+
+// String returns the string representation of the payment gateway type
+func (p PaymentGatewayType) String() string {
+	return string(p)
+}
+
+// WebhookEventType represents the type of webhook event
+type WebhookEventType string
+
+const (
+	// Stripe webhook events
+	WebhookEventTypeCheckoutSessionCompleted             WebhookEventType = "checkout.session.completed"
+	WebhookEventTypeCheckoutSessionAsyncPaymentSucceeded WebhookEventType = "checkout.session.async_payment_succeeded"
+	WebhookEventTypeCheckoutSessionAsyncPaymentFailed    WebhookEventType = "checkout.session.async_payment_failed"
+	WebhookEventTypeCheckoutSessionExpired               WebhookEventType = "checkout.session.expired"
+	WebhookEventTypeCustomerCreated                      WebhookEventType = "customer.created"
+	WebhookEventTypePaymentIntentPaymentFailed           WebhookEventType = "payment_intent.payment_failed"
+	WebhookEventTypePaymentIntentSucceeded               WebhookEventType = "payment_intent.succeeded"
+)
+
+// Validate validates the webhook event type
+func (w WebhookEventType) Validate() error {
+	switch w {
+	case WebhookEventTypeCheckoutSessionCompleted,
+		WebhookEventTypeCheckoutSessionAsyncPaymentSucceeded,
+		WebhookEventTypeCheckoutSessionAsyncPaymentFailed,
+		WebhookEventTypeCheckoutSessionExpired,
+		WebhookEventTypeCustomerCreated,
+		WebhookEventTypePaymentIntentPaymentFailed,
+		WebhookEventTypePaymentIntentSucceeded:
+		return nil
+	default:
+		return ierr.NewError("invalid webhook event type").
+			WithHint("Please provide a valid webhook event type").
+			WithReportableDetails(map[string]any{
+				"allowed": []WebhookEventType{
+					WebhookEventTypeCheckoutSessionCompleted,
+					WebhookEventTypeCheckoutSessionAsyncPaymentSucceeded,
+					WebhookEventTypeCheckoutSessionAsyncPaymentFailed,
+					WebhookEventTypeCheckoutSessionExpired,
+					WebhookEventTypeCustomerCreated,
+					WebhookEventTypePaymentIntentPaymentFailed,
+					WebhookEventTypePaymentIntentSucceeded,
+				},
+			}).
+			Mark(ierr.ErrValidation)
+	}
+}
+
+// String returns the string representation of the webhook event type
+func (w WebhookEventType) String() string {
+	return string(w)
+}
+
+// GetGatewayFromEventType returns the payment gateway type from a webhook event type
+func (w WebhookEventType) GetGatewayFromEventType() PaymentGatewayType {
+	switch w {
+	case WebhookEventTypeCheckoutSessionCompleted,
+		WebhookEventTypeCheckoutSessionAsyncPaymentSucceeded,
+		WebhookEventTypeCheckoutSessionAsyncPaymentFailed,
+		WebhookEventTypeCheckoutSessionExpired,
+		WebhookEventTypeCustomerCreated,
+		WebhookEventTypePaymentIntentPaymentFailed,
+		WebhookEventTypePaymentIntentSucceeded:
+		return PaymentGatewayTypeStripe
+	default:
+		return PaymentGatewayTypeStripe // Default to Stripe for unknown events
+	}
+}
```

Comment:
this file will move into payment gateway specific files in the payment gateway module